### DISCUSSION TITLE
Hermes doesn't work on Ubuntu 18.04

### DISCUSSION THREAD

**Discussion by web3cook** [informalsystems/hermes#1933] [Q&A] [answered]
When I try to use hermes binary on ubuntu18.04 it gives me the following error:
```
./hermes: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not found (required by ./hermes)
```

## Version

v0.12.0

## Steps to Reproduce

Just try running it on ubuntu18.04 x86_64.

____

## For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate milestone (priority) applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned

**Comment by romac** [comment, marked as answer]
The binary release of Hermes for Linux is built on Ubuntu 20.04 and links with glibc 2.29, which is more recent than the version of glibc that ships with Ubuntu 18.04, hence the error.

You should be able to build Hermes yourself on Ubuntu 18.04 by following the instructions in the guide: https://hermes.informal.systems/installation.html#install-via-cargo

This requires an installation of the Rust toolchain, for which you can find instructions at https://www.rust-lang.org/tools/install

Alternatively, you can try updating glibc to >=2.29, but I cannot provide instructions for doing so.

**Comment by romac** [comment]
Thanks for the report by the way, we will update the guide to mention the prerequisite on glibc 2.29.

**Comment by web3cook** [comment]
It works if I build it manually.
